libdrgn/type.c

```diff
@@ -1108,6 +1193,12 @@ static struct drgn_error *drgn_type_eq_impl(struct drgn_type *a,
 	if (drgn_type_has_is_variadic(a) &&
 	    drgn_type_is_variadic(a) != drgn_type_is_variadic(b))
 		goto out_false;
+	if (drgn_type_has_template_parameters(a) ||
+	    drgn_type_has_template_parameters(b)) {
```

Comment:
The types must be of the same kind at  this point, so it's redundant to also check `b`.